iptstate.cc

```diff
@@ -1091,24 +1093,36 @@ int conntrack_hook(enum nf_conntrack_msg_type nf_type, struct nf_conntrack *ct,
     return NFCT_CB_CONTINUE;
   }
 
-  if (flags->filter_src && (memcmp(&(entry->src), &(filters->src), entrysize))) {
-    counts->skipped++;
-    return NFCT_CB_CONTINUE;
+  if (flags->filter_src) {
+    if ((flags->filter_inv && !memcmp(&(entry->src), &(filters->src), entrysize)) || 
```

Comment:
Awesome, much cleaner. Would you mind fixing the indentation though? I can't quite tell if there's a tab snuck in there or you just indented differently... but line up the line-broken conditionals like so:

```
if (flags->filter_src) {
  if ((flags->filter_inv ...)) ||
      (!flags->filter_inv... ))) {
    counts->skipped++
    return NFCT_CB_CONTINUE;
  }
}
```

(If you don't feel inclined to do it I'll eventually just merge this and then fix it up in a commit after that)